tests/scorers/test_ragas_scorer.py

```diff
@@ -11,56 +11,63 @@
 )
 
 
-# Mock the create function
+# Mock the acompletion function
 @pytest.fixture
-def mock_create(monkeypatch):
-    def _mock_create(*args, **kwargs):
-        # Retrieve the response_model to return appropriate mock responses
-        response_model = kwargs.get("response_model")
-        if response_model is EntityExtractionResponse:
-            return EntityExtractionResponse(entities=["Paris"])
-        elif response_model is RelevancyResponse:
-            return RelevancyResponse(
-                reasoning="The context directly answers the question.",
-                relevancy_score=1,
-            )
+def mock_acompletion(monkeypatch):
+    async def _mock_acompletion(*args, **kwargs):
+        response_format = kwargs.get("response_format")
+        if response_format is EntityExtractionResponse:
+            content = '{"entities": ["Paris"]}'
+        elif response_format is RelevancyResponse:
+            content = '{"reasoning": "The context directly answers the question.", "relevancy_score": 1}'
 
-    monkeypatch.setattr("weave.scorers.ragas_scorer.create", _mock_create)
+        class Message(BaseModel):
+            content: str
+
+        class Choice(BaseModel):
+            message: Message
+
+        class Response(BaseModel):
+            choices: list[Choice]
+
+        return Response(choices=[Choice(message=Message(content=content))])
+
+    monkeypatch.setattr("weave.scorers.ragas_scorer.acompletion", _mock_acompletion)
 
 
 @pytest.fixture
-def context_entity_recall_scorer(mock_create):
+def context_entity_recall_scorer(mock_acompletion):
     return ContextEntityRecallScorer(
-        client=OpenAI(api_key="DUMMY_API_KEY"),
         model_id="gpt-4o",
         temperature=0.7,
         max_tokens=1024,
     )
 
 
 @pytest.fixture
-def context_relevancy_scorer(mock_create):
+def context_relevancy_scorer(mock_acompletion):
     return ContextRelevancyScorer(
-        client=OpenAI(api_key="DUMMY_API_KEY"),
         model_id="gpt-4o",
         temperature=0.7,
         max_tokens=1024,
     )
 
 
-def test_context_entity_recall_scorer_score(context_entity_recall_scorer):
+@pytest.mark.asyncio
+async def test_context_entity_recall_scorer_score(context_entity_recall_scorer):
     output = "Paris is the capital of France."
     context = "The capital city of France is Paris."
-    result = context_entity_recall_scorer.score(output, context)
+    result = await context_entity_recall_scorer.score(output, context)
     assert isinstance(result, dict)
     assert "recall" in result
     assert result["recall"] == 1.0  # Assuming full recall in mock response
```

Comment:
The last line implies all 3 right?

Reply:
yeah, but you get better error messages if we keep them separately, and we will refactor this to output Pydantic models anyway